components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/PreferenceRetrievalClient.java

```diff
@@ -0,0 +1,263 @@
+     *
+     * @param tenant tenant domain name.
+     * @return returns true if  question based password recovery enabled.
+     * @throws PreferenceRetrievalClientException
+     */
+    public Boolean checkQuestionBasedPasswordRecovery(String tenant) throws PreferenceRetrievalClientException {
+
+        return checkPreference(tenant, RECOVERY_CONNECTOR, QUESTION_PASSWORD_RECOVERY_PROPERTY);
+    }
+
+    /**
+     * Check password recovery is enabled or not.
+     *
+     * @param tenant tenant domain name.
+     * @return returns true if password recovery enabled.
+     * @throws PreferenceRetrievalClientException
+     */
+    public Boolean checkPasswordRecovery(String tenant) throws PreferenceRetrievalClientException {
+
+        List<String> propertyNameList = new ArrayList<String>();
+        propertyNameList.add(NOTIFICATION_PASSWORD_RECOVERY_PROPERTY);
+        propertyNameList.add(QUESTION_PASSWORD_RECOVERY_PROPERTY);
+        return checkMultiplePreference(tenant, RECOVERY_CONNECTOR, propertyNameList);
+    }
+
+    /**
+     * Check for preference in the given tenant.
+     *
+     * @param tenant        tenant domain name.
+     * @param connectorName name of the connector.
+     * @param propertyName  property to check.
+     * @return returns true if the property is enabled.
+     * @throws PreferenceRetrievalClientException
+     */
+    public Boolean checkPreference(String tenant, String connectorName, String propertyName)
+            throws PreferenceRetrievalClientException {
+
+        try (CloseableHttpClient httpclient = HttpClientBuilder.create().useSystemProperties().build()) {
+            JSONArray main = new JSONArray();
+            JSONObject preference = new JSONObject();
+            preference.put(CONNECTOR_NAME, connectorName);
+            JSONArray properties = new JSONArray();
+            properties.put(propertyName);
+            preference.put(PROPERTIES, properties);
+            main.put(preference);
+            HttpPost post = new HttpPost(getUserGovernancePreferenceEndpoint(tenant));
+            setTempAuthorizationHeader(post);
+            post.setEntity(new StringEntity(main.toString(), ContentType.create(HTTPConstants
+                    .MEDIA_TYPE_APPLICATION_JSON, Charset.forName(StandardCharsets.UTF_8.name()))));
+
+            try (CloseableHttpResponse response = httpclient.execute(post)) {
+
+                if (response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+                    JSONArray jsonResponse = new JSONArray(
+                            new JSONTokener(new InputStreamReader(response.getEntity().getContent())));
+                    JSONObject connector = (JSONObject) jsonResponse.get(0);
+                    JSONArray responseProperties = connector.getJSONArray(PROPERTIES);
+                    for (int itemIndex = 0, totalObject = responseProperties.length(); itemIndex < totalObject; itemIndex++) {
```

Comment:
Can responseProperties be null?

Reply:
No, but it can become an empty array if f we didn't require any property for a connector.